zkSNACKs/WalletWasabi
# On shutdown prevention, stop the CJM to participate in new rounds. On cancel, restart them.

Fixes: https://github.com/zkSNACKs/WalletWasabi/issues/10653

This PR stops all wallets in `InputReg phase` on shutdown prevention.
Also, it signals to the CJM not to participate in new rounds, once they are finished with the `Critical phase`.

If the user cancels the shutdown prevention (changed his mind about quitting), restart the CJs.

How to test:

1.  Do CJs with 4-5 wallets and make sure they are participating in 2-3 different rounds parallel.
2. Try to shutdown the app when one of the wallets in Critical Phase.
3. See that wallets in `input reg phase` are stopped/cancelled.
4. See that wallets in `Critical phase` don't participate in new rounds.
5. See that the app can shut down after every wallet leaves the Critical phase.

You can also test the case when you cancel the shutdown prevention, the wallets that were in input reg are restarted.
